frame/democracy/src/lib.rs

```diff
@@ -1739,10 +1740,11 @@ impl<T: Config> Pallet<T> {
 
 		// pick out another public referendum if it's time.
 		if (now % T::LaunchPeriod::get()).is_zero() {
-			// Errors come from the queue being empty. we don't really care about that, and even if
-			// we did, there is nothing we can do here.
-			let _ = Self::launch_next(now);
-			weight = max_block_weight;
+			// Errors come from the queue being empty. If the queue is not empty, it will take
+			// full block weight.
+			if Self::launch_next(now).is_ok() {
+				weight = max_block_weight;
+			}
```

Comment:
What we should do here is actually create a benchmark of how much time it takes to process an empty queue, and use that weight here.

You are right that this is probably a huge overestimate, but not recording any weight at all is probably worse than this.

Are you familiar with how do make the benchmark, and are you interested to complete that?

Reply:
completed.